jmxtrans-output/jmxtrans-output-gelf/src/test/java/com/googlecode/jmxtrans/model/output/gelf/GelfWriterFactoryTests.java

```diff
@@ -0,0 +1,332 @@
+import com.googlecode.jmxtrans.model.ValidationException;
+import org.assertj.core.api.Condition;
+import org.graylog2.gelfclient.GelfConfiguration;
+import org.graylog2.gelfclient.GelfTransports;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class GelfWriterFactoryTests {
+	@Test
+	public void testTcpDefaultsWriterFactory() throws ValidationException {
+
+		final ImmutableList<String> typenames = ImmutableList.of();
+		final Map<String, Object> settings = new HashMap<>();
+
+		final GelfWriterFactory gelfWriterFactory = new GelfWriterFactory(
+			typenames,
+			true,
+			true,
+			settings,
+			"test",
+			null,
+			ImmutableMap.of("test", (Object) "test"),
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null
+		);
+
+		assertThat(
+			gelfWriterFactory.getGelfConfiguration().getHostname()
+		)
+			.isEqualTo("test")
+			.as("Hostname is wrong");
+
+		// Check, that default values were used.
+
+		assertThat(gelfWriterFactory.getGelfConfiguration()).is(
+			new GelfDefaultConfigurationCondition()
+		).as("Values are not the expected default values");
+
+		final GelfWriter gelfWriter = gelfWriterFactory.create();
+
+		assertThat(gelfWriter)
+			.isNotNull()
+			.as("Didn't get a writer back.");
+		assertThat(gelfWriter.getAdditionalFields())
+			.containsEntry("test", "test")
+			.as("Wrong additional fields set");
+	}
+
+	@Test
+	public void testUdpDefaultsWriterFactory() throws ValidationException {
+
+		final ImmutableList<String> typenames = ImmutableList.of();
+		final Map<String, Object> settings = new HashMap<>();
+
+		final GelfWriterFactory gelfWriterFactory = new GelfWriterFactory(
+			typenames,
+			true,
+			true,
+			settings,
+			"test",
+			null,
+			ImmutableMap.of("test", (Object) "test"),
+			"udp",
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null,
+			null
+		);
+
+		assertThat(
+			gelfWriterFactory.getGelfConfiguration().getTransport()
+		)
+			.isEqualTo(GelfTransports.UDP)
+			.as("Wrong transport selected");
+
+		final GelfWriter gelfWriter = gelfWriterFactory.create();
+
+		assertThat(gelfWriter).isNotNull().as("Didn't get a writer back.");
+
+	}
+
+	@Test
+	public void testNonDefaultsWriterFactory() throws ValidationException {
+
+		final ImmutableList<String> typenames = ImmutableList.of();
+		final Map<String, Object> settings = new HashMap<>();
+
+		final GelfWriterFactory gelfWriterFactory = new GelfWriterFactory(
+			typenames,
+			true,
+			true,
+			settings,
+			"test",
+			12202,
+			ImmutableMap.of("test", (Object) "test"),
+			"udp",
+			10,
+			10,
+			10,
+			true,
+			10,
+			true,
+			"/somefile",
+			true,
+			true,
+			10
+		);
+
+		assertThat(
+			gelfWriterFactory.getGelfConfiguration()
+		).is(
+			new Condition<GelfConfiguration>() {
```

Comment:
Hm. I found the (rather long) series of assertThat not really readable. Maybe move it to another class as I did with GelfDefaultConfigurationCondition?